scripts/xe-toolstack-restart

```diff
@@ -24,40 +26,35 @@ fi
 echo "Executing $FILENAME"
 
 POOLCONF=`cat @ETCDIR@/pool.conf`
-SERVICES="xapi v6d squeezed perfmon xenopsd xenopsd-xc xenopsd-xenlight xenopsd-simulator xenopsd-libvirt xcp-rrdd-plugins xcp-rrdd-gpumon xcp-rrdd xcp-networkd fe forkexecd mpathalert-daemon xapi-storage-script"
+SERVICES="perfmon xapi v6d xenopsd xenopsd-xc xenopsd-xenlight xenopsd-simulator xenopsd-libvirt xcp-rrdd-iostat xcp-rrdd-squeezed xcp-rrdd-xenpm xcp-rrdd-gpumon xcp-rrdd xcp-networkd squeezed forkexecd mpathalert-daemon xapi-storage-script"
 
 TO_RESTART=""
 for svc in $SERVICES ; do
-	# xcp-networkd should be ignored if the do-not-use file exists
-	if [ $svc == xcp-networkd -a -e /tmp/do-not-use-networkd ] ; then
-		continue
-	fi
-
-	# mpathalert-daemon isn't in chkconfig, so needs to be handled specially
+	# mpathalert-daemon isn't a systemd service so needs to be handled specially
 	if [ $svc == mpathalert-daemon -a $POOLCONF == "master" ] ; then
 		TO_RESTART="$svc $TO_RESTART"
 		@ETCDIR@/master.d/03-mpathalert-daemon stop
 		continue
 	fi
 
-	# restart other services only if chkconfig said they were enabled
-	chkconfig $svc
+	# restart other services only if systemd said they were enabled
+	systemctl is-enabled $svc >/dev/null 2>&1
 
 	if [ $? -eq 0 ] ; then
 		TO_RESTART="$svc $TO_RESTART"
-		service $svc stop
+		action "systemctl stop $svc" systemctl stop $svc
```

Comment:
Can this whole loop be replaced by `system-ctl try-restart $SERVICES`? I suspect that systemd will then take care of the ordering.

> try-restart PATTERN...
> Restart one or more units specified on the command line if the units are running. This does nothing if units are not running.